core/src/smartcontracts/isi/mod.rs

```diff
@@ -80,25 +80,31 @@ impl Execute for RegisterBox {
         Span::current().record("id", &object_id.to_string());
         iroha_logger::trace!(%authority, "Executing");
         match object_id {
-            RegistrableBox::Peer(peer) => Register::<Peer>::new(*peer).execute(authority, wsv),
-            RegistrableBox::Domain(domain) => {
-                Register::<Domain>::new(*domain).execute(authority, wsv)
+            RegistrableBox::Peer(object) => {
+                Register::<Peer> { object: *object }.execute(authority, wsv)
```

Comment:
Yes, please do, I think with `Constructor` we can save some code space

Reply:
> Why so? Why not to use `new()`?

also these structs are only containers, there is no constructor specific code that should be run when creating them. It's also unlikely users would be constructing them because they are only created when unboxing.

Reply:
it's not possible to destructure a `Box`

Reply:
I missed that it's a box.